File: demo/test-samples.js
// Sample JavaScript code for testing AI Assistant extension features

// Email validation function - good for explaining, documenting, testing
function validateEmail(email) {
    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
    return emailRegex.test(email);
}

// Problematic function - good for problem finding
function divide(a, b) {
    return a / b; // Missing zero check, no input validation
}

// Function that could be refactored - good for refactoring suggestions
function getUserInfo(userId) {
    var users = getUsers();
    var user = null;
    for (var i = 0; i < users.length; i++) {
        if (users[i].id == userId) {
            user = users[i];
            break;
        }
    }
    if (user != null) {
        return user.name + " (" + user.email + ")";
    } else {
        return "User not found";
    }
}

// Simple function for unit test generation
function calculateArea(radius) {
    return Math.PI * radius * radius;
}

// Class for documentation and testing
class UserManager {
    constructor() {
        this.users = [];
    }
    
    addUser(user) {
        this.users.push(user);
    }
    
    findUser(id) {
        return this.users.find(u => u.id === id);
    }
}

// Complex regex for explanation testing
const complexEmailRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;

// SQL query for explanation testing
const getUserQuery = `
    SELECT u.id, u.name, u.email, p.title as position
    FROM users u
    LEFT JOIN positions p ON u.position_id = p.id
    WHERE u.active = 1
    ORDER BY u.created_at DESC
    LIMIT 10
`;

// Cron expression for explanation testing
const dailyBackupCron = "0 2 * * *"; // Every day at 2 AM

// Function with poor naming - good for name suggestions
function fn1(x, y) {
    return x * y + 10;
}

// Incomplete function - good for code generation
function processData(data) {
    // TODO: Implement data processing logic
    // Should validate input, transform data, and return results
}

File: demo/test-samples.py
# Sample Python code for language conversion testing

import re
from typing import List, Optional

def validate_email(email: str) -> bool:
    """
    Validates an email address using regex pattern matching
    """
    email_regex = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
    return bool(re.match(email_regex, email))

class UserManager:
    """
    A simple user management class
    """
    def __init__(self):
        self.users = []
    
    def add_user(self, user: dict) -> None:
        """Add a user to the manager"""
        self.users.append(user)
    
    def find_user(self, user_id: int) -> Optional[dict]:
        """Find a user by ID"""
        for user in self.users:
            if user.get('id') == user_id:
                return user
        return None
    
    def get_active_users(self) -> List[dict]:
        """Get all active users"""
        return [user for user in self.users if user.get('active', False)]

# Function with potential issues
def divide_numbers(a, b):
    return a / b  # No zero division check

# Sample data for testing
sample_users = [
    {"id": 1, "name": "John Doe", "email": "john@example.com", "active": True},
    {"id": 2, "name": "Jane Smith", "email": "jane@example.com", "active": False},
    {"id": 3, "name": "Bob Johnson", "email": "bob@example.com", "active": True}
]